### PR TITLE
Task 8 profile back end

### PR DESCRIPTION
**Pull request by universezzz**

### CODE REVIEW

**Review by OlegPanasyuk — APPROVED** [review]

File: api/controllers/profile.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const services = require('../services/profile');
+
+const { publicProfileValidate } = require('../validation/profile');
+
+const { isEmpty } = require('lodash');
```
Comment (melonholy) [inline api/controllers/profile.js]: import native libraries at the beginning

File: api/services/profile.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+const db = require('../../database');
+const errors = require('./errorHandlers/index');
+const jwt_decode = require('jwt-decode');
+
+const getProfile = async token => {
+  const { id } = jwt_decode(token);
```
Comment (melonholy) [inline api/services/profile.js]: if future we need to write middleware for token decoding and we are using `jsonwebtoken` library for working with jwt tokens

File: package.json [hunk 3]
```diff
@@ -20,8 +20,9 @@
     "express": "^4.17.1",
     "express-async-errors": "^3.1.1",
     "jsonwebtoken": "^8.5.1",
-    "nodemailer": "^6.3.0",
+    "jwt-decode": "^2.2.0",
```
Comment (melonholy) [inline package.json:23]: we already have `jsonwebtoken` as a main library for working with jwt tokens

File: api/services/profile.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    linkedInLink,
+    facebookLink,
+    description,
+  };
+
+  return profile;
+};
+
+const updateProfile = async req => {
+  const { id } = jwt_decode(req.headers.token);
```
Comment (melonholy) [inline api/services/profile.js]: comment above

**Review by melonholy — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (universezzz) [reply package.json:23]: has removed.

File: api/services/profile.js (hunk 4 above)
Comment (universezzz) [reply api/services/profile.js]: has fixed

File: api/services/profile.js (hunk 2 above)
Comment (universezzz) [reply api/services/profile.js]: has fixed

File: api/controllers/profile.js (hunk 1 above)
Comment (universezzz) [reply api/controllers/profile.js]: has fixed

**Review by slavasgod — APPROVED** [review]

**Pull request merged**